wrappers/rust/icicle-curves/icicle-bn254/src/poseidon2/mod.rs

```diff
@@ -0,0 +1,35 @@
+use crate::curve::{ScalarCfg, ScalarField};
+
+use icicle_core::error::IcicleResult;
+use icicle_core::impl_poseidon2;
+use icicle_core::poseidon2::{DiffusionStrategy, MdsType, Poseidon2, Poseidon2Config, Poseidon2Constants};
+use icicle_core::traits::IcicleResultWrap;
+use icicle_cuda_runtime::device_context::DeviceContext;
+use icicle_cuda_runtime::error::CudaError;
+use icicle_cuda_runtime::memory::HostOrDeviceSlice;
```

Comment:
Can we move these to the poseidon2 mod inside the `impl_poseidon2` macro?

Reply:
we don't do that for any other primitive. So I guess not

Reply:
Ok, i guess its out of scope for this PR but its just more confusing boilerplate in each field/curve crate that can be moved to the core crate

Reply:
Yes, I totally agree with you